File: example-android/app/src/main/java/com/hedera/android_example/ui/main/SectionsPagerAdapter.java
package com.hedera.android_example.ui.main;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class SectionsPagerAdapter extends FragmentStateAdapter {
    public SectionsPagerAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position) {
            case 0:
                return new PrivateKeyFragment();
            case 1:
                return new AccountBalanceFragment();
            case 2:
                return new CryptoTransferFragment();
            default:
                throw new IllegalArgumentException();
        }
    }

    @Override
    public int getItemCount() {
        return 3;
    }
}

File: example-android/app/src/main/java/com/hedera/android_example/ui/main/SectionsPagerAdapter.kt
/*-
 *
 * Hedera Java SDK
 *
 * Copyright (C) 2022 - 2024 Hedera Hashgraph, LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */
package com.hedera.android_example.ui.main

import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import androidx.fragment.app.Fragment
import java.lang.IllegalArgumentException

class SectionsPagerAdapter(fragmentActivity: FragmentActivity) :
    FragmentStateAdapter(fragmentActivity) {
    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> PrivateKeyFragment()
            1 -> AccountBalanceFragment()
            2 -> CryptoTransferFragment()
            else -> throw IllegalArgumentException()
        }
    }

    override fun getItemCount(): Int {
        return 3
    }
}
